Updated POM and all Jelly Files

nisarg1499 opened the pull request:
Development Environment:

Maven Version - 3.6.0
JDK Version - 1.8.0_171
OS - Windows10
I tried to run discard-old-build-plugin on my machine using mvn clean install command, but InjectedTest failure occurred. That test is not designed for Windows. And so I updated the POM to the recent version and done few required changes.

## **Proposed Changelog enteries**
-  Updated Version
- Removed dom4j
- Added Jenkins version and java level
- Removed distributionManagement
- Added harmcrest-core 1.3
- Changed mokito-all to mokito-core
- HTTPs done in repositories and pluginRepositories

Find discussion [here](https://github.com/jenkinsci/discard-old-build-plugin/pull/9)

## **Desired Reviewers**
@oleg-nenashev @BenjaminBeggs

oleg-nenashev commented:
What is the difference from https://github.com/jenkinsci/discard-old-build-plugin/pull/9 ?

nisarg1499 commented:
> What is the difference from #9 ?

There is no difference. I have also referenced [PR](https://github.com/jenkinsci/discard-old-build-plugin/pull/9) in the description.
Actually I was working on some other branch, but due to mistake I made all changes in branch used in [this](https://github.com/jenkinsci/discard-old-build-plugin/pull/9) PR.
So I made a new branch and created a new PR.

nisarg1499 commented:
@BenjaminBeggs  I have made `commit 13222c6` (latest) which will resolve the conflicts.

nisarg1499 commented:
@BenjaminBeggs I think now this can be integrated.

BenjaminBeggs commented:
One quick thing, what are the implications of these version changes on backwards compatibility with older Jenkins instances? There are some legacy Jenkins systems that I’d like to maintain compatibility with, if possible. @oleg-nenashev

martinda commented:
Hi. I have a legacy jenkins instance that cannot be upgraded (1.6xx series). I am hoping that the simple fix of decision logic in the discard decision can be made without upgrading to the latest jenkins. After that, those upgrades can be made.

nisarg1499 commented:
> Hi. I have a legacy jenkins instance that cannot be upgraded (1.6xx series). I am hoping that the simple fix of decision logic in the discard decision can be made without upgrading to the latest jenkins. After that, those upgrades can be made.

What type of fix in decision logic are you expecting?

BenjaminBeggs commented:
The fix in decision logic is a feature enhancement I will be posting as a PR shortly. I am going to wait on merging this as I work on getting the plugin to pass its automation test while maintaining compatibility with legacy Jenkins instances.

nisarg1499 commented:
> The fix in decision logic is a feature enhancement I will be posting as a PR shortly. I am going to wait on merging this as I work on getting the plugin to pass its automation test while maintaining compatibility with legacy Jenkins instances.

Okay.
And I have committed the new change(added MIT license and BenjaminBegs as developer) in my PR, still merge conflicts are there. Any specific thing I am missing?

BenjaminBeggs commented:
I'm not sure the remaining conflicts, you would need to check a diff between the master branch files and the PR files.

nisarg1499 commented:
> I'm not sure the remaining conflicts, you would need to check a diff between the master branch files and the PR files.

@BenjaminBeggs 
Well I checked the diff between master branch files and PR files. No diff is there and only the required changes are their in PR files.
I think you should revert the PR in which you have added MIT license and your name as developer. First merge this PR and then add the reverted PR. And anyways I have added MIT license and your name as a developer.

Actually at first when you  added MIT license at that time there was a  conflict. And so in my PR, I added MIT license commit, so now conflicts should be gone.

BenjaminBeggs commented:
I will take care of the conflicts.

nisarg1499 commented:
> I will take care of the conflicts.

Okay.
According to me, now there is no conflict in pom file. So it can be processed further.

nisarg1499 commented:
> > I will take care of the conflicts.
> 
> Okay.
> According to me, now there is no conflict in pom file. So it can be processed further.

Also I have shared results of these new POM [here](https://github.com/jenkinsci/discard-old-build-plugin/pull/9).

BenjaminBeggs commented:
I’ll get to this soon, don’t be concerned if it isn’t merge immediately. I’ll be in touch if I need anything from you ;).

nisarg1499 commented:
> I’ll get to this soon, don’t be concerned if it isn’t merge immediately. I’ll be in touch if I need anything from you ;).

Okay.

BenjaminBeggs commented:
I have resolved your merge conflict. Your PR does not pass the Jenkins build check, however.

nisarg1499 commented:
> I have resolved your merge conflict. Your PR does not pass the Jenkins build check, however.

@oleg-nenashev 
Yes there is some error regarding `Non-resolvable parent POM for jp.shiftinc.jenkins.plugins:discard-old-build:1.06-SNAPSHOT: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version and 'parent.relativePath' points at wrong local POM @ line 3, column 11`.
I will have a look on it and try to solve

nisarg1499 commented:
@oleg-nenashev @BenjaminBeggs 
I think that this can be solved by giving jenkins version 3.43 in properties, and increasing version of mockito-core and hamcrest. I will do commit tomorrow. 
Do let me know if you have any thoughts.

BenjaminBeggs commented:
I’ll keep an eye open for your commits! Good luck :)

BenjaminBeggs commented:
From what I see you're still struggling to use HTTPS for repositories, correct? @oleg-nenashev do you have advice on this? I too have reverted to using HTTP, which seems to still be a common practice for many active Jenkins plugins.

nisarg1499 commented:
> From what I see you're still struggling to use HTTPS for repositories, correct? @oleg-nenashev do you have advice on this? I too have reverted to using HTTP, which seems to still be a common practice for many active Jenkins plugins.

I don't know. But may be reverting to HTTP can work.

nisarg1499 commented:
> From what I see you're still struggling to use HTTPS for repositories, correct? @oleg-nenashev do you have advice on this? I too have reverted to using HTTP, which seems to still be a common practice for many active Jenkins plugins.

But from the error message I don't think that there is some issue regarding HTTPs.

BenjaminBeggs commented:
I received similar (the same?) error messages when using HTTPS, which went away solely by reverting to HTTP. I checked the practices followed by some active and popular Jenkins plugins and found HTTP still used commonly. If you get this PR passing build tests while using HTTPS I will be interested to see how. :)

nisarg1499 commented:
@BenjaminBeggs 
Reverted to HTTP, but still same issue.

nisarg1499 commented:
@BenjaminBeggs @oleg-nenashev 
Any suggestions for this error

Maybe we are referring to a non existing POM, some problem is occurring due to parent pom.
```
[2019-06-08T10:25:11.725Z] [ERROR] [ERROR] Some problems were encountered while processing the POMs:

[2019-06-08T10:25:11.725Z] [FATAL] Non-resolvable parent POM for jp.shiftinc.jenkins.plugins:discard-old-build:1.06-SNAPSHOT: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version and 'parent.relativePath' points at wrong local POM @ line 3, column 11

[2019-06-08T10:25:11.725Z]  @ 

[2019-06-08T10:25:11.725Z] [ERROR] The build could not read 1 project -> [Help 1]

[2019-06-08T10:25:11.725Z] org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:

[2019-06-08T10:25:11.725Z] [FATAL] Non-resolvable parent POM for jp.shiftinc.jenkins.plugins:discard-old-build:1.06-SNAPSHOT: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version and 'parent.relativePath' points at wrong local POM @ line 3, column 11

[2019-06-08T10:25:11.725Z] 

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:383)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.725Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.725Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.725Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.725Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.725Z] [ERROR]   

[2019-06-08T10:25:11.725Z] [ERROR]   The project jp.shiftinc.jenkins.plugins:discard-old-build:1.06-SNAPSHOT (C:\Jenkins\workspace\s_discard-old-build-plugin_PR-11\pom.xml) has 1 error

[2019-06-08T10:25:11.725Z] [ERROR]     Non-resolvable parent POM for jp.shiftinc.jenkins.plugins:discard-old-build:1.06-SNAPSHOT: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version and 'parent.relativePath' points at wrong local POM @ line 3, column 11 -> [Help 2]

[2019-06-08T10:25:11.725Z] org.apache.maven.model.resolution.UnresolvableModelException: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:197)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:243)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally (DefaultModelBuilder.java:1052)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParent (DefaultModelBuilder.java:830)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:332)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:432)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:400)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:363)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.725Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.725Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.725Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.725Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.725Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.725Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.725Z] Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version

[2019-06-08T10:25:11.725Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:422)

[2019-06-08T10:25:11.725Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:224)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:201)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:260)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:193)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:243)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally (DefaultModelBuilder.java:1052)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParent (DefaultModelBuilder.java:830)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:332)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:432)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:400)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:363)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.726Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.726Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.726Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.726Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.726Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.726Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.726Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.726Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.726Z] Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact org.jenkins-ci.plugins:plugin:pom:3.43 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Received fatal alert: protocol_version

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed (ArtifactTransportListener.java:52)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:365)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute (BasicRepositoryConnector.java:583)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get (BasicRepositoryConnector.java:259)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads (DefaultArtifactResolver.java:498)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:399)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:224)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:201)

[2019-06-08T10:25:11.726Z]     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:260)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:193)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:243)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally (DefaultModelBuilder.java:1052)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParent (DefaultModelBuilder.java:830)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:332)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:432)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:400)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:363)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.726Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.983Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.983Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.983Z] Caused by: org.apache.maven.wagon.TransferFailedException: Received fatal alert: protocol_version

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.wagon.shared.AbstractHttpClientWagon.fillInputData (AbstractHttpClientWagon.java:1069)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.wagon.shared.AbstractHttpClientWagon.fillInputData (AbstractHttpClientWagon.java:963)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.getInputStream (StreamWagon.java:126)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.getIfNewer (StreamWagon.java:88)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.get (StreamWagon.java:61)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run (WagonTransporter.java:567)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter.execute (WagonTransporter.java:435)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter.get (WagonTransporter.java:412)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask (BasicRepositoryConnector.java:453)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:360)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute (BasicRepositoryConnector.java:583)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get (BasicRepositoryConnector.java:259)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads (DefaultArtifactResolver.java:498)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:399)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:224)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:201)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:260)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:193)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:243)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally (DefaultModelBuilder.java:1052)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParent (DefaultModelBuilder.java:830)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:332)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:432)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:400)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:363)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.983Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.983Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.983Z] Caused by: javax.net.ssl.SSLException: Received fatal alert: protocol_version

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.Alerts.getSSLException (Alerts.java:208)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.Alerts.getSSLException (Alerts.java:154)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.SSLSocketImpl.recvAlert (SSLSocketImpl.java:1979)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.SSLSocketImpl.readRecord (SSLSocketImpl.java:1086)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.SSLSocketImpl.performInitialHandshake (SSLSocketImpl.java:1332)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.SSLSocketImpl.startHandshake (SSLSocketImpl.java:1359)

[2019-06-08T10:25:11.983Z]     at sun.security.ssl.SSLSocketImpl.startHandshake (SSLSocketImpl.java:1343)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket (SSLConnectionSocketFactory.java:396)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.conn.ssl.SSLConnectionSocketFactory.connectSocket (SSLConnectionSocketFactory.java:355)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.conn.DefaultHttpClientConnectionOperator.connect (DefaultHttpClientConnectionOperator.java:142)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.conn.PoolingHttpClientConnectionManager.connect (PoolingHttpClientConnectionManager.java:373)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.MainClientExec.establishRoute (MainClientExec.java:381)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.MainClientExec.execute (MainClientExec.java:237)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.ProtocolExec.execute (ProtocolExec.java:185)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec.execute (RetryExec.java:89)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RedirectExec.execute (RedirectExec.java:111)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.client.InternalHttpClient.doExecute (InternalHttpClient.java:185)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.httpclient.impl.client.CloseableHttpClient.execute (CloseableHttpClient.java:83)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.wagon.shared.AbstractHttpClientWagon.execute (AbstractHttpClientWagon.java:825)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.wagon.shared.AbstractHttpClientWagon.fillInputData (AbstractHttpClientWagon.java:986)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.providers.http.wagon.shared.AbstractHttpClientWagon.fillInputData (AbstractHttpClientWagon.java:963)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.getInputStream (StreamWagon.java:126)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.getIfNewer (StreamWagon.java:88)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.wagon.StreamWagon.get (StreamWagon.java:61)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run (WagonTransporter.java:567)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter.execute (WagonTransporter.java:435)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.transport.wagon.WagonTransporter.get (WagonTransporter.java:412)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask (BasicRepositoryConnector.java:453)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:360)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute (BasicRepositoryConnector.java:583)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get (BasicRepositoryConnector.java:259)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads (DefaultArtifactResolver.java:498)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:399)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:224)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:201)

[2019-06-08T10:25:11.983Z]     at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:260)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:193)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.ProjectModelResolver.resolveModel (ProjectModelResolver.java:243)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally (DefaultModelBuilder.java:1052)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.readParent (DefaultModelBuilder.java:830)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:332)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:432)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:400)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:363)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.collectProjects (DefaultGraphBuilder.java:414)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.getProjectsForMavenReactor (DefaultGraphBuilder.java:405)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.graph.DefaultGraphBuilder.build (DefaultGraphBuilder.java:82)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.buildGraph (DefaultMaven.java:507)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:219)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

[2019-06-08T10:25:11.983Z]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

[2019-06-08T10:25:11.983Z]     at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:57)

[2019-06-08T10:25:11.983Z]     at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

[2019-06-08T10:25:11.983Z]     at java.lang.reflect.Method.invoke (Method.java:606)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

[2019-06-08T10:25:11.983Z]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

[2019-06-08T10:25:11.983Z] [ERROR] 

[2019-06-08T10:25:11.983Z] [ERROR] Re-run Maven using the -X switch to enable full debug logging.

```

BenjaminBeggs commented:
The solution to your issue is not immediately clear to me, though you can review my PR#14 to see what the issue may be. My PR implements most of the good practice recommendations yours does, although again omits the use of HTTPS at present. Mine passes the unit tests.

nisarg1499 commented:
> The solution to your issue is not immediately clear to me, though you can review my PR#14 to see what the issue may be. My PR implements most of the good practice recommendations yours does, although again omits the use of HTTPS at present. Mine passes the unit tests.

Well I checked, and didn't found anything wrong. I can't understand why mine is not passing unit tests.

BenjaminBeggs commented:
The answer to your question must lie somewhere in a careful analysis of the differential between my PR#14 POM.xml and your POM.xml. That being said, my PR#14 addresses all the things yours does except I further updated the dependencies the plugin uses and have omitted the use of HTTPS for the plugin repositories. When I merge PR#14, there will be no need for this one.
As such, I will be closing this PR now.
